multichain-api-gateway/src/lib.rs

```diff
@@ -45,58 +46,82 @@ impl From<BlockscoutSettings> for ApiEndpoints {
 }
 
 impl ApiEndpoints {
-    async fn make_requests(self, query: &str) -> Vec<(Instance, String)> {
-        let client = Client::new();
+    async fn make_requests(
+        self,
+        query: &str,
+        method: &Method,
+        body: Bytes,
+        headers: &HeaderMap,
+    ) -> Vec<(Instance, String)> {
+        let client = Client::builder()
+            .timeout(std::time::Duration::from_secs(60))
+            .finish();
 
         stream::iter(self.apis)
-            .map(|(instance, mut url)| async {
+            .map(|(instance, mut url)| {
                 url.set_query(Some(query));
-                let resp = client.get(url).send().await.unwrap();
-                (instance, resp.bytes().await)
+                (instance, url.to_string())
             })
-            .buffer_unordered(self.concurrent_requests)
-            .map(|(instance, response)| match response {
-                Ok(bytes) => (
-                    instance,
-                    str::from_utf8(bytes.as_ref()).unwrap().to_string(),
-                ),
-                Err(e) => (instance, e.to_string()),
+            .map(|(instance, url)| async {
+                let mut request = client.request(method.clone(), url);
+                let request_headers = request.headers_mut();
+                for (key, value) in headers {
+                    request_headers.insert(key.clone(), value.clone());
+                }
+                let response = request.send_body(body.clone()).await;
+                let str_response = match response {
+                    Ok(mut response) => match response.body().await {
+                        Ok(bytes) => match str::from_utf8(bytes.as_ref()) {
+                            Ok(str) => str.to_string(),
+                            Err(utf8_error) => utf8_error.to_string(),
+                        },
+                        Err(payload_error) => payload_error.to_string(),
+                    },
+                    Err(send_request_error) => send_request_error.to_string(),
+                };
+                (instance, str_response)
             })
+            .buffer_unordered(self.concurrent_requests)
             .collect()
             .await
     }
 }
 
 type Responses = HashMap<String, HashMap<String, Value>>;
 
-fn merge_responses(json_resonses: Vec<(Instance, String)>) -> Responses {
+fn merge_responses(json_responses: Vec<(Instance, String)>) -> Responses {
     let mut result: Responses = HashMap::new();
 
-    json_resonses
+    json_responses
         .into_iter()
         .for_each(|(Instance(net, subnet), value)| {
             let kv_subnet = result.entry(net).or_insert_with(HashMap::new);
             kv_subnet.insert(
                 subnet,
-                serde_json::from_str(&value).unwrap_or_else(|e| Value::String(e.to_string())),
+                serde_json::from_str(&value)
+                    .unwrap_or_else(|e| Value::String(format!("{}: {}\n", e, value))),
```

Comment:
Before that, when the server returned an error, the error was replaced with`expected value at column 1 at row 1` (json parsing failed). Now we preserve both errors